x-pack/libbeat/management/api/configuration.go

```diff
@@ -96,25 +97,43 @@ func (c *AuthClient) Configuration() (ConfigBlocks, error) {
 		blocks[block.Type] = append(blocks[block.Type], &ConfigBlock{Raw: block.Raw})
 	}
 
+	// keep the ordering consistent while grouping the items.
+	keys := make([]string, len(blocks))
+	idx := 0
+	for k := range blocks {
+		keys[idx] = k
+		idx++
+	}
+	sort.Strings(keys)
+
 	res := ConfigBlocks{}
-	for t, b := range blocks {
+	for _, t := range keys {
+		fmt.Printf("eeekeys? %+v\n", t)
+		b := blocks[t]
 		res = append(res, ConfigBlocksWithType{Type: t, Blocks: b})
 	}
 
 	return res, nil
 }
 
 // ConfigBlocksEqual returns true if the given config blocks are equal, false if not
-func ConfigBlocksEqual(a, b ConfigBlocks) bool {
-	if len(a) != len(b) {
-		return false
+func ConfigBlocksEqual(a, b ConfigBlocks) (bool, error) {
+	// If there is an errors when hashing the config blocks its because the format changed.
+	aHash, err := hashstructure.Hash(a, nil)
+	if err != nil {
+		return false, errors.Wrap(err, "could not hash config blocks")
+	}
+
+	bHash, err := hashstructure.Hash(b, nil)
+	if err != nil {
+		return false, errors.Wrap(err, "could not hash config blocks")
 	}
 
-	if len(a) == 0 {
-		return true
+	if aHash != bHash {
+		return false, nil
 	}
 
-	return reflect.DeepEqual(a, b)
+	return true, nil
```

Comment:
I wonder why `golint` or `go vet` do not complain here, but: `return aHash == bHash, nil`.

Reply:
I would prefer a hash in the long term, especially if we could diminish the roundtrip of call, I would be OK to implement our own hashing strategies later, I think for know hashstructure's Hash is an OK solution and would take care of a few false positive refresh.